mindmeld/components/dialogue.py

```diff
@@ -914,15 +914,26 @@ def _prompt_slot(self, responder, nlr):
         self._retry_attempts = 0
         self._prompt_turn = False
 
-    def _retry_logic(self, responder, nlr):
+    def _retry_logic(self, request, responder, nlr):
         if self._retry_attempts < self._max_retries:
             self._retry_attempts += 1
             responder.reply(nlr)
         else:
             # max attempts exceeded, reset counter, exit auto_fill.
             self._retry_attempts = 0
             self._exit_flow(responder)
-            self._app.app_manager.dialogue_manager.reprocess()
+
+            processed_query = self._app.app_manager.nlp.process(query_text=request.text)
+
+            request = self._app.app_manager.request_class(
+                context=request.context or {},
+                history=request.history or [],
+                frame=responder.frame or {},
+                params=FrozenParams(**DialogueResponder.to_json(responder.params)),
```

Comment:
request.params?

Reply:
No, we're replicating the responder params to become the new request params. Basically what Conversation.process would do in the app_manager.

The idea is the replicate the next turn without actually going to the next turn.

Reply:
`params=FrozenParams(**responder.params.to_dict()),`